l10n_it_fatturapa_in_notifications/account.py

```diff
@@ -19,11 +19,42 @@
 ##############################################################################
 
 from openerp.osv import fields, orm
+from openerp.tools.translate import _
+import netsvc
 
 
 class AccountInvoice(orm.Model):
     _inherit = "account.invoice"
     _columns = {
         'result_notification_id': fields.many2one(
-            'fatturapa.notification', "Result notification", readonly=True)
+            'fatturapa.notification', "Result notification", readonly=True),
+        'rejected': fields.boolean('Rejected')
         }
+
+    _defaults = {
+        'rejected': False
+    }
+
+    def action_cancel_draft(self, cr, uid, ids, *args):
+        if not hasattr(ids, '__iter__'):
+            ids = [ids]
+        for inv_id in ids:
+            invoice = self.browse(cr, uid, inv_id)
+            if invoice.rejected:
+                raise orm.except_orm(
+                    _('Attention'),
+                    _('Is not possible remove an invoice rejected')
+                )
+            else:
+                return super(
```

Comment:
non direi.. 
semplicemente gli passo sempre una lsta con un solo item per tutti gli id in modo da non ripetere 2 volte lo stesso ciclo...
vedi:
super(AccountInvoice, self).action_cancel_draft(cr, uid, [inv_id], *args)